### ISSUE TITLE
neil does not find a newer version of a library that antq does

### ISSUE THREAD

**Issue by bombaywalla** [babashka/neil#171]
I have an alias `:outdated` that runs `antq`.
`antq` finds a newer version of the library that `neil dep upgrade` does not.
Even though they both seem to use `version-clj`.
Admittedly, the name of this library wrt versioning is a bit on strange side.
A list of all the versions of this lib on Maven Central is at [this link](https://central.sonatype.com/artifact/com.google.apis/google-api-services-sheets/v4-rev20221216-2.0.0/versions)

Below is a small repro example.

```bash
dorabs-imac:test dorab$ cat deps.edn
{:paths ["src" "resources"]
 :deps {org.clojure/clojure {:mvn/version "1.11.1"}
        com.google.apis/google-api-services-sheets {:mvn/version "v4-rev20220927-2.0.0"}
        }
}

dorabs-imac:test dorab$ clojure -M:outdated 
[##################################################] 2/2

|    :file |                                      :name |             :current |              :latest |
|----------+--------------------------------------------+----------------------+----------------------|
| deps.edn | com.google.apis/google-api-services-sheets | v4-rev20220927-2.0.0 | v4-rev20221216-2.0.0 |

dorabs-imac:test dorab$ neil dep upgrade --dry-run
dorabs-imac:test dorab$ 
```

**Comment by borkdude** [comment]
The issue is that there is a check for "stable version" which currently is implemented as "any qualifier like -foo indicates a non-stable version". This should probably be fine-tuned, maybe by filtering on "alpha", "beta", etc.

**Comment by borkdude** [comment]
Should be fixed with 0.1.58
